userspace/plugin/plugin_types.h

```diff
@@ -128,6 +128,17 @@ typedef struct ss_plugin_byte_buffer {
 	const void* ptr;
 } ss_plugin_byte_buffer;
 
+// Used in extract_fields_and_offsets to receive field offsets along
+// with field data.
+// Extraction functions should be set these to the zero-indexed start
+// and end offsets of the field in the event or log data. 0..0 can be
+// used to indicate that there are no valid offsets, e.g. if the field
+// was generated or computed from other data.
```

Comment:
We need to support the case where the plugin doesn't know or care about extracting offsets at all. IMO we should document that the plugin can just leave the fields untouched (because every <3.11 plugin and many 3.11+ will) and it's the caller's responsibility to zero-initialize the buffer.

Then the 0..0 convention would be mostly (only?) handled by the caller (if they want to mark invalid ranges as u32::MAX..u32::MAX, feel free to do so)